### PR TITLE
lightbulb: Update to version 2.3

### PR DESCRIPTION
**Pull request by linsui**

### CODE REVIEW

File: bucket/lightbulb.json [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
     ],
     "persist": "Configuration.dat",
-    "notes": "If the gamma doesn't change under 6600K, run '$dir\\Manual_Gamma_Registry_Fix.reg'.",
     "checkver": "github",
     "autoupdate": {
-        "url": "https://github.com/Tyrrrz/LightBulb/releases/download/$version/LightBulb-Portable.zip"
-    }
+        "url": "https://github.com/Tyrrrz/LightBulb/releases/download/$version/LightBulb.zip"
+    },
+    "notes": "In order to run LightBulb 2.0 and future versions you will need to download and install .NET Core 3 runtime (follow the steps for \"Run desktop apps\")."
```
Comment (Ash258) [inline bucket/lightbulb.json]: - No URl, no anything ...
- This is really odd when command line installer tell you to follow some steps on some webpage

```suggestion
    "notes": "In order to run LightBulb 2.0 and future versions you will need to download and install .NET Core 3 runtime (follow the steps for 'Run desktop apps')."
```

File: bucket/lightbulb.json [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
-            "LightBulb_Portable.bat",
+            "LightBulb.exe",
             "LightBulb",
-            "",
-            "LightBulb.exe"
+            "-portable"            
         ]
     ],
     "persist": "Configuration.dat",
-    "notes": "If the gamma doesn't change under 6600K, run '$dir\\Manual_Gamma_Registry_Fix.reg'.",
```
Comment (Ash258) [inline bucket/lightbulb.json]: Why this was removed? It was fixed or??

Comment (linsui) [reply bucket/lightbulb.json]: > Added a check at startup that will verify that extended gamma range is enabled on your computer and, if not, prompt you to enable it automatically. This also means that portable distributions of LightBulb no longer require you to change registry yourself.

See https://github.com/Tyrrrz/LightBulb/releases/tag/2.0

File: bucket/lightbulb.json (hunk 1 above)
Comment (linsui) [reply bucket/lightbulb.json]: How about `In order to run LightBulb 2.0 and future versions you will need to download and install .NET Core 3 runtime (follow the steps for 'Run desktop apps' on https://dotnet.microsoft.com/download/dotnet-core/3.0/runtime ).` ?

Comment (linsui) [reply bucket/lightbulb.json]: It seems that lightbulb will bundle .Net Core in the future so we can also just wait. See https://github.com/Tyrrrz/LightBulb/issues/122#issuecomment-557826926

**Comment by rasa** [comment]
#2308

File: bucket/lightbulb.json (hunk 1 above)
Comment (linsui) [reply bucket/lightbulb.json]: Now the app can detect .Net runtime and download it, so I remove the notes. Please review it.

File: bucket/lightbulb.json [hunk 3]
```diff
@@ -1,29 +1,26 @@
...
-            "LightBulb_Portable.bat",
-            "LightBulb",
-            "",
-            "LightBulb.exe"
+            "LightBulb.exe",
+            "LightBulb"
         ]
     ],
-    "persist": "Configuration.dat",
+    "persist": "Settings.json",
```
Comment (Ash258) [inline bucket/lightbulb.json:21]: Do migration or at least notify users, that you are going to throw all of theirs configurations

File: bucket/lightbulb.json [hunk 4]
```diff
@@ -1,29 +1,26 @@
...
+    "description": "Computer screen gamma adjuster based on the current time",
     "homepage": "https://github.com/Tyrrrz/LightBulb",
     "license": "GPL-3.0-or-later",
-    "notes": "If the gamma doesn't change under 6600K, run '$dir\\Manual_Gamma_Registry_Fix.reg'.",
-    "url": "https://github.com/Tyrrrz/LightBulb/releases/download/1.6.4.1/LightBulb-Portable.zip",
-    "hash": "15ba9a6a9c890da8a787dc7af041349b66e179821ab7a28dc115f80c458853c4",
-    "pre_install": "if (!(Test-Path \"$persist_dir\\Configuration.dat\")) { Set-Content \"$dir\\Configuration.dat\" -Value '{}' -Encoding Ascii | Out-Null }",
+    "url": "https://github.com/Tyrrrz/LightBulb/releases/download/2.3/LightBulb.zip",
+    "hash": "402cbd6e206713ea13ca3097acaa9d36b96f200236eb70bd3deb8bd26ba371b2",
+    "pre_install": "if (!(Test-Path \"$persist_dir\\Settings.json\")) { Set-Content \"$dir\\Settings.json\" -Value '{\"IsAutoUpdateEnabled\": false}' -Encoding Ascii }",
```
Comment (Ash258) [inline bucket/lightbulb.json:8]:
```suggestion
    "pre_install": "if (!(Test-Path \"$persist_dir\\Settings.json\")) { Set-Content \"$dir\\Settings.json\" -Value '{ \"IsAutoUpdateEnabled\": false }' -Encoding Ascii }",
```

**Review by Ash258 — CHANGES_REQUESTED** [review]
Rename PR accordingly

File: bucket/lightbulb.json (hunk 3 above)
Comment (linsui) [reply bucket/lightbulb.json:26]: Though the old Configuration.dat is also in json format, all items have different names. So I only add a notification.

File: bucket/lightbulb.json [hunk 5]
```diff
@@ -1,29 +1,31 @@
...
     "license": "GPL-3.0-or-later",
-    "notes": "If the gamma doesn't change under 6600K, run '$dir\\Manual_Gamma_Registry_Fix.reg'.",
-    "url": "https://github.com/Tyrrrz/LightBulb/releases/download/1.6.4.1/LightBulb-Portable.zip",
-    "hash": "15ba9a6a9c890da8a787dc7af041349b66e179821ab7a28dc115f80c458853c4",
-    "pre_install": "if (!(Test-Path \"$persist_dir\\Configuration.dat\")) { Set-Content \"$dir\\Configuration.dat\" -Value '{}' -Encoding Ascii | Out-Null }",
+    "url": "https://github.com/Tyrrrz/LightBulb/releases/download/2.3/LightBulb.zip",
+    "hash": "402cbd6e206713ea13ca3097acaa9d36b96f200236eb70bd3deb8bd26ba371b2",
+    "pre_install": [
+        "if (!(Test-Path \"$persist_dir\\Settings.json\")) {",
+        "    if (Test-Path \"$persist_dir\\Configuration.dat\") { Write-Host 'Old configuration is not migrated!' -f Yellow }",
```
Comment (Ash258) [inline bucket/lightbulb.json:10]:
```suggestion
        "    if (Test-Path \"$persist_dir\\Configuration.dat\") { warn 'Configuration from version 1 is not compatile with version 2!' }",
```

**Review by Ash258 — CHANGES_REQUESTED** [review]
There is no user friendly message or anything which indicate .net runtime issue.
Suggest windows desktop runtim

image

**Review by rasa — APPROVED** [review]

**Review by Ash258 — APPROVED** [review]

**Pull request merged**